### PR TITLE
VPN-5545 - Remove all tutorial related code from tree

### PR DESCRIPTION
**Pull request by brizental**
Tutorials are not a supported feature since v2.17. We should be safe to remove all code related to it by now.

### CODE REVIEW

**Review by strseb — APPROVED** [review]
r+

**Comment by MattLichtenstein** [comment]
Removing the tutorials from the repo will eventually lead to them not being shipped to old version of the client. Do we want that? Not having tutorials on those old clients may have poor/unintended affects on the Tips and Tricks UI

**Comment by strseb** [comment]
>Removing the tutorials from the repo will eventually lead to them not being shipped to old version of the client. Do we want that? Not having tutorials on those old clients may have poor/unintended affects on the Tips and Tricks UI. 

That would only affect people now installing "old" versions of the client, (right?) we cannot "unship" them for users who have them i thought 😅

**Comment by MattLichtenstein** [comment]
> That would only affect people now installing "old" versions of the client, (right?) we cannot "unship" them for users who have them i thought 😅

Maybe we should double check, but I believe we remove addons from the system when they are no longer in the manifest

**Comment by MattLichtenstein** [comment]
@strseb looks like we move unknown addons here https://github.com/mozilla-mobile/mozilla-vpn-client/blob/b2ea208aa49de20a67b272a06f4aeda3cf9381e0/src/addons/manager/addonmanager.cpp#L107-L126

**Review by lesleyjanenorton — CHANGES_REQUESTED** [review]
### ✨ 🧹🧹🧹 ✨
Very shiny!
While we are at it we should also remove:
- logo-error.svg
- logo-success.scvg
- logo-warning.svg

There is some cleanup we should do in `ExternalOpHandler` (maybe we no longer even need it? IIRC it was implemented to catch external operations that could interrupt a tutorial).

**Comment by lesleyjanenorton** [comment]
> Removing the tutorials from the repo will eventually lead to them not being shipped to old version of the client. Do we want that? Not having tutorials on those old clients may have poor/unintended affects on the Tips and Tricks UI

@MattLichtenstein good point. Looping in @albionx. I'll add it to tomorrow's agenda.

File: src/commands/commandui.cpp [hunk 1]
```diff
@@ -51,6 +51,7 @@
 #include "settingsholder.h"
 #include "telemetry.h"
 #include "temporarydir.h"
+#include "theme.h"
```
[the remark this answers is not in the document]
Comment (brizental) [reply src/commands/commandui.cpp:54]: Great question, probably unnecessary 🤔

**Comment by brizental** [comment]
> There is some cleanup we should do in ExternalOpHandler (maybe we no longer even need it? IIRC it was implemented to catch external operations that could interrupt a tutorial).

This is correct, the only registered blocker for the external op handler was tutorials. That is not gone on: [9bed839](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/8534/commits/9bed8399861ed71ba950ca7f59f4d4c912daba9e)

File: src/commands/commandui.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (brizental) [reply src/commands/commandui.cpp:54]:
```suggestion
```

**Review by lesleyjanenorton — APPROVED** [review]
Nice, thank you!

**Comment by albionx** [comment]
Provided we can't avoid that older clients lose the tutorials, I think we should wait before merging this in. I just checked on Looker and ~43% of clients are on versions 2.16 and below. Maybe we revisit in Mid/Late Dec, once 2.19 is out?

**Comment by brizental** [comment]
> Provided we can't avoid that older clients lose the tutorials, I think we should wait before merging this in. I just checked on Looker and ~43% of clients are on versions 2.16 and below. Maybe we revisit in Mid/Late Dec, once 2.19 is out?

I will just leave the addons manifests in there. I'll go ahead with the removal of the code. cc @lesleyjanenorton

**Pull request merged**